matrix_client/crypto/olm_device.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+import olm
+from canonicaljson import encode_canonical_json
+
+from matrix_client.checks import check_user_id
+
+logger = logging.getLogger(__name__)
+
+
+class OlmDevice(object):
+    """Manages the Olm cryptographic functions.
+
+    Has a unique Olm account which holds identity keys.
+
+    Args:
+        api (MatrixHttpApi): The api object used to make requests.
+        user_id (str): Matrix user ID. Must match the one used when logging in.
+        device_id (str): Must match the one used when logging in.
+    """
+
+    def __init__(self, api, user_id, device_id):
+        self.api = api
+        check_user_id(user_id)
+        self.user_id = user_id
+        self.device_id = device_id
+        self.olm_account = olm.Account()
+        self.olm_account.create()
```

Comment:
On creating the account, shouldn't the public part of the fingerprint key pair and identity key pair be uploaded as part of class instantiation (also checking/registering one-time keys)? Note that I'm reviewing commit-by-commit, so if this is added later on in the commit series, I apologize.

Reply:
Is this just because of the scope of this PR (limited to things that don't interact with C2S api)?

Reply:
It is added later, but not in __init__. It shouldn't be there because I don't think you would expect an instantiation to require network functionality, especially since the other classes of the SDK don't. Also it grants the freedom of creating a device offline, which may be useful to do before logging in, and is definitely very useful when testing.

Reply:
sounds reasonable